src/window_handle.rs

```diff
@@ -748,11 +774,40 @@ impl WindowHandle {
                         if cx.app_state.focus != Some(id) {
                             let old = cx.app_state.focus;
                             cx.app_state.focus = Some(id);
+
+                            // Fix: Focus events were never dispatched to views' own
+                            // event_before_children / event_after_children.
+                            //
+                            // This is a bit messy; could it be done in ViewId.apply_event()
+                            // instead?  Are there other cases of dropped events?
+                            let mut ec = EventCx {
+                                app_state: cx.app_state,
+                            };
+
+                            if let Some(old) = old.as_ref() {
+                                ec.unconditional_view_event(*old, Event::FocusLost, true);
+                            }
+                            ec.unconditional_view_event(id, Event::FocusGained, true);
+
+                            cx = UpdateCx {
+                                app_state: ec.app_state,
+                            };
+
                             cx.app_state.focus_changed(old, cx.app_state.focus);
```

Comment:
`unconditional_view_event` probably needs to called in `focus_changed`.

E.g. when the focus got changed due to pointer click.